src/airstack/generic.py

```diff
@@ -225,3 +234,465 @@ def remove_unused_variables(document_ast, query):
             if query.count(_variable.variable.name.value) == 1:
                 del document_ast.definitions[0].variable_definitions[_count]
     return print_ast(document_ast)
+
+
+def format_poaps_data(poaps, existing_user=None):
+    if existing_user is None:
+        existing_user = []
+
+    recommended_users = existing_user.copy()
+    for poap in poaps or []:
+        attendee = poap.get('attendee', {})
+        poap_event = poap.get('poapEvent', {})
+        event_id = poap.get('eventId')
+
+        name = poap_event.get('eventName')
+        content_value = poap_event.get('contentValue', {})
+        addresses = attendee.get('owner', {}).get('addresses', [])
```

Comment:
This will fail if`attendee = poap.get('attendee', {})` is None. You should check what the value type of `attendee` is before calling a get on it